frontend/packages/console-dynamic-plugin-sdk/src/api/dynamic-core-api.ts

```diff
@@ -128,3 +121,19 @@ export const InventoryItemLoading: React.FC = require('@console/shared/src/compo
   .InventoryItemLoading;
 
 export { useFlag } from '../utils/flags';
+
+/**
+ * A lazy component that creates a monaco editor
+ * @example
+ * ```tsx
+ * <ResourceYAMLEditor
+ *     content={yamlString}
+ *     header="Create resource"
+ *    />
+ * ```
+ * @param {string} content - Stringified YAML/Object to be shown by the Editor
+ * @param {string} header - Add a custom header for the YAML editor
+ *
+ */
+export const ResourceYAMLEditor: React.FC<EditYAMLProps> = require('@console/internal/components/AsyncYamlEditor')
+  .DroppableEditYAML;
```

Comment:
Won't this expose the other props on `DroppableEditYAML` (even if not documented or part of the TypeScript type)? It would be better to lock this down so plugins can't pass those props. I'm afraid plugins will use those props and later get broken.

Reply:
Added a wrapper to lock props.